src/api/PDFPageOptions.ts

```diff
@@ -178,3 +179,44 @@ export interface PDFPageDrawSVGElementOptions {
   fonts?: { [fontName: string]: PDFFont };
   blendMode?: BlendMode;
 }
+
+/**
+ * Options for adding an annotation to a PDF page.
+ */
+export interface PDFPageAddAnnotationOptions {
+  subtype: AnnotationTypes;
+  rect: { x: number; y: number; width: number; height: number };
+  contents?: string;
+  name?: string;
+  flags?: number;
+  color?: number[];
+  border?: number[];
+  modificationDate?: Date;
+}
+
+/**
+ * Options for adding a markup annotation to a PDF page.
+ *
+ * Markup annotations refers:
+ * - Highlight
+ * - Underline
+ * - Squiggly
+ * - StrikeOut
+ * annotations.
+ */
+export interface PDFPageAddTextMarkupAnnotationOptions
+  extends PDFPageAddAnnotationOptions {
+  /**
+   * The quad points that define the region(s) to be marked up.
+   */
+  quadPoints: {
```

Comment:
That's a very annoying data format. Please use Rect instead. Way easier to manipulate!